pkg/cloudprovider/launchtemplate.go

```diff
@@ -102,6 +104,40 @@ func (p *LaunchTemplateProvider) Get(ctx context.Context, provider *v1alpha1.AWS
 	if provider.LaunchTemplateName != nil {
 		return map[string][]cloudprovider.InstanceType{ptr.StringValue(provider.LaunchTemplateName): nodeRequest.InstanceTypeOptions}, nil
 	}
+	options, err := p.createAmiOptions(ctx, provider, lo.Assign(nodeRequest.Template.Labels, additionalLabels))
+	if err != nil {
+		return nil, err
+	}
+	resolvedLaunchTemplates, err := p.amiFamily.Resolve(ctx, provider, nodeRequest, options)
+	if err != nil {
+		return nil, err
+	}
+	launchTemplates := map[string][]cloudprovider.InstanceType{}
+	for _, resolvedLaunchTemplate := range resolvedLaunchTemplates {
+		// Ensure the launch template exists, or create it
+		ec2LaunchTemplate, err := p.ensureLaunchTemplate(ctx, resolvedLaunchTemplate)
+		if err != nil {
+			return nil, err
+		}
+		launchTemplates[*ec2LaunchTemplate.LaunchTemplateName] = resolvedLaunchTemplate.InstanceTypes
+	}
+	return launchTemplates, nil
+}
+
+func (p *LaunchTemplateProvider) GetAmisForProvider(ctx context.Context, provider *v1alpha1.AWS, providerRef *v1alpha5.ProviderRef, instanceTypes []cloudprovider.InstanceType) ([]string, error) {
+	if provider.LaunchTemplateName != nil {
+		return nil, fmt.Errorf("using a custom Launch Template which is deprecated")
+	}
+	options, err := p.createAmiOptions(ctx, provider, map[string]string{})
+	if err != nil {
+		return nil, err
+	}
+	amifamily := amifamily.GetAMIFamily(provider.AMIFamily, options)
+	amiIds, err := p.amiFamily.AmiProvider.Get(ctx, providerRef, options, instanceTypes, amifamily)
```

Comment:
Why is this `p.amiFamily` and not `amiFamily` directly? It looks like you aren't using the correct family here.

Reply:
Ohh, sorry for the confusion. But `p.amiFamily` is actually `*amifamily.Resolver` type. Would you mind if I rename `p.amiFamily` to `p.amiResolver` ?